docker/pristine/Dockerfile

```diff
@@ -0,0 +1,8 @@
+### Used for pristine builds
+#
+ARG GOVERSION=latest
+FROM golang:${GOVERSION}
```

Comment:
It seems like this build is refactored out of the `docker run` (which you modified above). I'm seeing that it has a Golang version of `{GO_DOCKER_VERSION}`. I'm curious about using the latest version here.

Reply:
ARG specifies build-arguments you can pass in and you can specify default values for them. In this case I have GOVERSION as an argument I can pass in with `latest` as the default (which is just the latest golang docker image). This lets me run it without having to always worry about specifying GOVERSION while keeping the ability to specify a specific version.

We need GOVERSION specified as there is always a bit of delay after release when everything used (docker, circleci, etc). sync up with the new release. It also keeps the version nailed down and upgrading it as a commit (allowing bisecting issues introduced by the new go version).

Hope I answered what you were curious about in there somewhere. :)

Reply:
thank you, yes! that makes sense - latest is just the default value and another variable still can contain a specified value.